### PR TITLE
[Pages] Main Page Framework & Goods & Category Page

### PR DESCRIPTION
**Pull request by YXHXianYu**
closes: #2
closes: #3
closes: #4

### CODE REVIEW

File: frontend/src/pages/Main/index.tsx [hunk 1]
```diff
@@ -1,11 +1,99 @@
...
+    setCurrentIndex((prevIndex) => (prevIndex - 1 + ids.length) % ids.length);
+  };
+
+  const good = createAsync(() => getGood(ids[currentIndex()]));
+
+  // 自动翻页
+  createEffect(() => {
+    const intervalId = setInterval(nextImage, 3000); // 每3秒翻页一次
+    onCleanup(() => clearInterval(intervalId)); // 组件卸载时清除定时器
+  });
```
Comment (AzurIce) [inline frontend/src/pages/Main/index.tsx:40]: maybe should use [onMount](https://docs.solidjs.com/concepts/effects#onmount)?

Comment (YXHXianYu) [reply frontend/src/pages/Main/index.tsx:40]: Let me have a try (

File: frontend/src/assets/MinatoAqua2.jpg
Comment (AzurIce) [inline frontend/src/assets/MinatoAqua2.jpg:1]: 吸溜（

File: frontend/src/pages/Categories/CategoriesWrapper.tsx
Comment (AzurIce) [inline frontend/src/pages/Categories/CategoriesWrapper.tsx:1]: TODO: add a category menu to choose category

**Pull request merged**